cardano-cli/src/Cardano/CLI/Types/Common.hs

```diff
@@ -389,6 +390,14 @@ instance ToJSON SlotsTillKesKeyExpiry where
 instance FromJSON SlotsTillKesKeyExpiry where
   parseJSON v = SlotsTillKesKeyExpiry <$> parseJSON v
 
+
+data TxOutShelleyBasedEra
+  = TxOutShelleyBasedEra
+      !(Address ShelleyAddr)
+      Value
+      TxOutDatumAnyEra
+      ReferenceScriptAnyEra
+  deriving Show
```

Comment:
We should exclude `ShelleyBasedEra` from all type and function names and have all the Byron specific types and functions of carrying the burden of qualifying that it is only valid for `ByronOnly`.

Reply:
Yep :+1: . This naming is just to make it painfully obvious what I'm doing. Eventually these types won't have anything in its name to indicate the era.